backend/app/routers/listeners.py

```diff
@@ -23,6 +26,28 @@
 clowder_bucket = os.getenv("MINIO_BUCKET_NAME", "clowder")
 
 
+@router.get("/instance")
+async def get_instance_id(
+    user=Depends(get_current_user),
+    db: MongoClient = Depends(get_db),
+):
+    # Check all connection and abort if any one of them is not available
+    if db is None:
+        raise HTTPException(status_code=503, detail="Service not available")
+        return
+
+    if (instance_id := await db["config"].find_one({"key": "instance_id"})) is not None:
+        return ConfigEntryOut.from_mongo(instance_id).value
+    else:
+        # If no ID has been generated for this instance, generate a 10-digit alphanumeric identifier
+        instance_id = ''.join(random.choice(string.ascii_uppercase + string.ascii_lowercase + string.digits) for _ in range(10))
```

Comment:
is there a convention for the random id? should we use uuid or anything else? thoughts?

Reply:
not sure... this particular snippet is same logic Clowder 1 was using.